sqlmesh/core/model/definition.py

```diff
@@ -1876,6 +1877,21 @@ def _split_sql_model_statements(
     return query, expressions[:pos], expressions[pos + 1 :]
 
 
+def _resolve_custom_session_params(
+    defaults: t.Dict[str, t.Any] | None, **kwargs: t.Dict[str, t.Any]
+) -> None:
+    if kwargs.get("session_properties") and defaults and defaults.get("session_properties"):
+        session_properties = kwargs["session_properties"]
+        session_props = {expr.this.name for expr in session_properties}
+        for k, v in defaults["session_properties"].items():
+            if k not in session_props:
+                new_eq = exp.EQ(
+                    this=exp.Literal(this=k), expression=exp.Column(this=exp.Identifier(this=v))
```

Comment:
The value can also be a sqlglot expression, that's why I'd suggest doing:
```
this=exp.Literal(this=k), expression=exp.convert(v)
```
Also CC: @georgesittas @tobymao to correct me if I'm wrong.